src/App.js

```diff
@@ -1,19 +1,25 @@
 import React, { } from 'react'
-import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
+import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom'
 import 'bootstrap/dist/css/bootstrap.min.css'
 import './App.css'
 import Home from './scenes/Home/Home'
 import WeatherHourlyList from './scenes/WeatherHourlyList/WeatherHourlyList'
 import Login from './scenes/Login/Login'
+import { getToken } from './useToken'
 
 function App () {
   return (
     <div className='App'>
       <Router>
         <Switch>
-          <Route exact path='/' component={Home} />
-          <Route path='/login' component={Login } />
-          <Route exact path='/detail/:locationId/:year/:month/:day' component={WeatherHourlyList} />
+          <Route path='/login' component={Login} >
+            {getToken() ? <Redirect to='/' /> : <Login />}
+          </Route>
+          <Route exact path='/' component={Home}>
+            {getToken() ? <Home /> : <Redirect to='/login' />}
+            <Route path='/login' component={Login}></Route>
+          </Route>
+          <Route path='/detail/:locationId/:year/:month/:day' component={WeatherHourlyList} />
```

Comment:
Do you feel this is complicated?
Should implement this way.

- For the case unauthenticated, you can validate it in the App component by returning the login page if no token.
==> in App component, if no token, return login page
```js
function App () {
  if(!getToken()) {
    return <Login />
  }

  return (
    <div className='App'>
     ...
    </div>
  )
}
```

- For the case authenticated and go to log in, you can check in the Login component by redirecting to /home if there's a token.
==> in Login component, if getToken ok, redirect to Home
```js
const Login = () => {
  if(getToken) {
    window.location.replace('/')
  }
}
```
but this will be checked every time the Login component re-render when users type a username or password. It should check only the first render. Then you should find a solution.

Reply:
> but this will be checked every time the Login component re-render when users type a username or password. It should check only the first render. Then you should find a solution.

Remember reading carefully ☝️